treescript/src/treescript/l10n.py

```diff
@@ -77,26 +81,82 @@ def build_platform_dict(bump_config, repo_path):
     return platform_dict
 
 
+# get_latest_revision {{{1
+async def get_latest_revision(locale, url):
+    """Download the hg pushlog for the latest locale revision.
+
+    Args:
+        locale (str): the locale to query
+        url (str): the [templatized] pushlog url
+
+    Returns:
+        tuple (locale, revision)
+
+    """
+    url = url % {"locale": locale}
+    with tempfile.NamedTemporaryFile() as fp:
+        path = fp.name
+        await retry_async(download_file, args=(url, path), retry_exceptions=(DownloadError,))
+        revision_info = load_json_or_yaml(path, is_path=True)
+    last_push_id = revision_info["lastpushid"]
+    revision = revision_info["pushes"][str(last_push_id)]["changesets"][0]
```

Comment:
Are these pushes guaranteed to only have one changeset? It might be safer to take `[-1]` instead of `[0]`.

Reply:
As long as we have `tipsonly=1` in the query args, that appears to be true.